.github/workflows/build_workflow.yml

```diff
@@ -0,0 +1,306 @@
+          docker compose -f docker-compose.yml -f docker-compose-e2e-tests.yml build gp2gp-e2e-tests
+          docker compose -f docker-compose.yml -f docker-compose-e2e-tests.yml up --exit-code-from mongodb activemq gp2gp wiremock gpcc gp2gp-e2e-tests
+        working-directory: ./docker
+
+      - name: Get End-to-End Test Reports
+        run: docker cp e2e-tests:/home/gradle/e2e-tests/build/reports/** .
+        working-directory: ./service/build/reports
+
+      - name: Dump Docker Logs
+        if: always()
+        run: |
+          chmod +x dump_docker_logs.sh
+          ./dump_docker_logs.sh
+        working-directory: ./scripts
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: 'End-to-End Test Results & Docker Logs'
+          path: |
+            ./service/build/reports/**
+            ./scripts/logs/**
+          compression-level: 9
+
+      - name: Stop Docker Dependencies
+        if: always()
+        run: |
+          docker compose -f docker/docker-compose.yml -f docker/docker-compose-e2e-tests.yml down
+          docker network rm commonforgp2gp
+        working-directory: ./docker
+
+# ------------------------------------------------------------
+
+  docker:
+    name: Build & Publish Docker Images
+    runs-on: ubuntu-latest
+    needs: [unit-tests, integration-tests, end-to-end-tests]
+    permissions:
+        id-token: write
+        contents: read
+    strategy:
+      matrix:
+        config:
+          - directory: service
+            repository: gp2gp
+            build-context: .
+          - directory: wiremock
+            repository: gp2gp-wiremock
+            build-context: .
+          - directory: mock-mhs-adaptor
+            repository: gp2gp-mock-mhs
+            build-context: .
+          - directory: gpcc-mock
+            repository: gp2gp-gpcc-mock
+            build-context: ./docker/gpcc-mock
+          - directory: gpc-api-mock
+            repository: gp2gp-gpc-api-mock
+            build-context: ./docker/gpc-api-mock
+          - directory: sds-api-mock
+            repository: gp2gp-sds-api-mock
+            build-context: ./docker/sds-api-mock
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          role-session-name: gp2gp_pull_request
+          aws-region: ${{ secrets.AWS_REGION }}
+
+      - name: Setup Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Generate Build ID
+        run: |
+          BUILD_ID=$(python3 ./scripts/tag.py ${{ github.head_ref || github.ref_name }} ${{ github.run_number }} ${{ github.sha }})
+          echo "Generated the build tag: $BUILD_ID"
+          echo "BUILD_ID=$BUILD_ID" >> $GITHUB_ENV
```

Comment:
Setting up Python to run a script feels overkill, let's port `tag.py` to bash?

Reply:
I wouldn't worry too much about this for now, as it's what's there already for Jenkins. Not like this PR is making the situation any worse.

Reply:
Agreed, we can revisit at a later stage for enhancements like this :)